### PR TITLE
Add ffill/bfill

### PR DESCRIPTION
**Pull request by milesgranger**

### CODE REVIEW

File: dask_expr/_expr.py [hunk 1]
```diff
@@ -2603,6 +2603,56 @@ def _fusion_pass(expr):
...
 
 
+class Fill(MapOverlap):
+    _parameters = [
+        "frame",
+        "how",
+        "axis",
+        "limit",
+    ]
+    _defaults = {"axis": 0, "limit": None}
```
Comment (phofl) [inline dask_expr/_expr.py:2613]: Let's remove axis for now

File: dask_expr/_expr.py [hunk 2]
```diff
@@ -2603,6 +2603,56 @@ def _fusion_pass(expr):
...
+    before = 1
+    after = 0
+    enforce_metadata = True
+
+    @staticmethod
+    def func(frame, how, **kwargs):
```
Comment (phofl) [inline dask_expr/_expr.py:2619]: I think you can use M.ffill as operation instead of how?

File: dask_expr/_expr.py [hunk 3]
```diff
@@ -2603,6 +2603,56 @@ def _fusion_pass(expr):
     return expr
 
 
+class Fill(MapOverlap):
+    _parameters = [
+        "frame",
+        "how",
```
Comment (phofl) [inline dask_expr/_expr.py:2609]: Can we specify a class variable how and do another BFill class?

File: dask_expr/_expr.py [hunk 4]
```diff
@@ -2603,6 +2603,56 @@ def _fusion_pass(expr):
     return expr
 
 
+class Fill(MapOverlap):
```
Comment (phofl) [inline dask_expr/_expr.py:2606]: FFill is a better name

Comment (milesgranger) [reply dask_expr/_expr.py]: https://github.com/dask-contrib/dask-expr/pull/454/commits/0b201d8a143e4b0981d32ef5e3ab52366fea5b2b

File: dask_expr/_expr.py (hunk 2 above)
Comment (milesgranger) [reply dask_expr/_expr.py]: https://github.com/dask-contrib/dask-expr/pull/454/commits/9381794c1e7d4948d4be0365f9bb1d7143a0b886

File: dask_expr/_expr.py (hunk 1 above)
Comment (milesgranger) [reply dask_expr/_expr.py]: Gone https://github.com/dask-contrib/dask-expr/pull/454/commits/1129903c787c0d19a0ae00513067642809adeafa

File: dask_expr/_expr.py [hunk 5]
```diff
@@ -2666,6 +2666,58 @@ def _fusion_pass(expr):
...
+            kwargs=self.kwargs,
+        )
+
+
+class BFill(FFill):
+    func = M.bfill
+
+    def _simplify_down(self):
+        mapoverlap = super()._simplify_down()
+        mapoverlap.before, mapoverlap.after = mapoverlap.after, mapoverlap.before
```
[the remark this answers is not in the document]
Comment (milesgranger) [reply dask_expr/_expr.py]: https://github.com/dask-contrib/dask-expr/pull/454/commits/5a0cf982574a3e23cc3dbbd636c62bc1077d3e34

File: dask_expr/_expr.py [hunk 6]
```diff
@@ -2672,6 +2672,66 @@ def _fusion_pass(expr):
...
+            enforce_metadata=self.enforce_metadata,
+            kwargs=self.kwargs,
+        )
+
+
+class BFill(FFill):
+    func = M.bfill
+
+    def _simplify_down(self):
+        self.after, self.before = 1 if self.limit is None else self.limit, 0
```
Comment (phofl) [inline dask_expr/_expr.py:2722]: I meant making after and before actual properties that return the correct value based on limit

Comment (milesgranger) [reply dask_expr/_expr.py]: https://github.com/dask-contrib/dask-expr/pull/454/commits/e8af6d6132c209c90f175703ce4126b98e25ff42

**Comment by milesgranger** [comment]
[Failure seems unrelated,](https://github.com/dask-contrib/dask-expr/actions/runs/7112078256/job/19361315706?pr=454#step:5:207) but cannot re-run it myself

**Comment by phofl** [comment]
thx

**Pull request merged**